curriculum/challenges/english/25-front-end-development/review-bash-scripting/6724e417419c2f211bb41bfc.md

```diff
@@ -7,7 +7,708 @@ dashedName: review-bash-scripting
 
 # --description--
 
-Description placeholder
+## Bash Scripting Basics
+
+- **Bash scripting**: Writing a sequence of Bash commands in a file, which you can then execute with Bash to run the contents of the file.
+- **Shebang**: The commented line at the beginning of a script (e.g., `#!/bin/bash`) that indicates what interpreter should be used for the script.
+
+  ```bash
+  #!/bin/bash
+  ```
+
+- **Variable assignment**: Instantiate variables using the syntax `variable_name=value`.
+
+  ```bash
+  servers=("prod" "dev")
+  ```
+
+- **Variable creation rules**: Create variables with `VARIABLE_NAME=VALUE` syntax. No spaces are allowed around the equal sign (`=`). Use double quotes if the value contains spaces.
+
+  ```bash
+  NAME=John
+  MESSAGE="Hello World"
+  COUNT=5
+  TEXT="The next number is, "
+  ```
+
+- **Variable usage**: Access variable values by placing `$` in front of the variable name.
+
+  ```bash
+  echo $NAME
+  echo "The message is: $MESSAGE"
+  ```
+
+- **Variable interpolation**: Use `$variable_name` to access the value of a variable within strings and commands.
+
+```bash
+TEXT="The next number is, "
+NUMBER=42
+echo $TEXT B:$NUMBER
+echo $TEXT I:$NUMBER
+
+echo "Pulling $server"
+rsync --archive --verbose $server:/etc/nginx/conf.d/server.conf configs/$server.conf
+```
```

Comment:
````suggestion
  ```bash
  TEXT="The next number is, "
  NUMBER=42
  echo $TEXT B:$NUMBER
  echo $TEXT I:$NUMBER
  
  echo "Pulling $server"
  rsync --archive --verbose $server:/etc/nginx/conf.d/server.conf configs/$server.conf
  ```
````